server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java

```diff
@@ -191,4 +195,16 @@ private SMInputFactory initStax() {
     xmlFactory.setProperty(XMLInputFactory.IS_VALIDATING, Boolean.FALSE);
     return new SMInputFactory(xmlFactory);
   }
+
+  private enum BackupActiveRuleComparator implements Comparator<ActiveRule> {
```

Comment:
Saw this in @sns-seb 's code and liked it immediately: the most concise way to declare a static singleton class with implementation in Java - see e.g http://stackoverflow.com/questions/26285520/implementing-singleton-as-enum-in-java